src/utils/OnboardingProvider.tsx

```diff
@@ -71,49 +73,49 @@ export default function OnboardingProvider({
      * Does not affect database.
      */
     const updateProfile = (updatedInfo: Partial<Profile>) => {
-      const newProfileData = { ...profile, ...updatedInfo };
-      setProfile(newProfileData);
+      const newUserProfile = { ...userProfile, ...updatedInfo };
+      setUserProfile(newUserProfile);
     };
```

Comment:
for this and all other functions in the `useMemo`, could you put them outside the `useMemo` and wrap each of them individually in a `useCallback`?

with the current configuration, all of these functions will be re-defined every time the `useMemo` is re-computed (which happens a lot) - ideally we'd just want to define functions once.